pkgs/development/python-modules/kfactory/default.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  setuptools_scm,
+  wheel,
+  klayout,
+  aenum,
+  cachetools,
+  gitpython,
+  loguru,
+  pydantic,
+  pydantic-settings,
+  rectangle-packer,
+  requests,
+  ruamel-yaml-string,
+  scipy,
+  tomli,
+  toolz,
+  typer,
+  numpy,
+  ruamel-yaml,
+}:
+
+buildPythonPackage rec {
+  pname = "kfactory";
+  version = "0.21.7";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "gdsfactory";
+    repo = "kfactory";
+    rev = "v${version}";
+    sha256 = "sha256-VLhAJ5rOBKEO1FDCnlaseA+SmrMSoyS+BaEzjdHm59Y=";
+  };
+
+  nativeBuildInputs = [
+    setuptools_scm
+    tomli
+    wheel
+  ];
+
+  pythonRemoveDeps = [
+    "klayout"
+  ];
+
+  buildInputs = [
+    klayout
+  ];
+
+  propagatedBuildInputs = [
+    aenum
+    cachetools
+    gitpython
+    loguru
+    numpy
+    pydantic
+    pydantic-settings
+    rectangle-packer
+    requests
+    ruamel-yaml
+    ruamel-yaml-string
+    scipy
+    tomli
+    toolz
+    typer
+  ];
+
+  postInstall = ''
+    ln -s "${klayout}/lib/pymod/klayout" "$out/lib/python$(python -V | sed -E 's/Python ([0-9]+\.[0-9]+).*/\1/')/site-packages/klayout"
+  '';
```

Comment:
Can we add `klayout = toPythonModule pkgs.klayout;` to `pkgs/top-level/python-packages.nix` instead?

Reply:
The `klayout`s python modules are not in the standard path, but at `$out/lib/pymod`. Is there a way to tell `toPythonModule` about this?

Reply:
@natsukium I'd appreciate an advise on this.

Reply:
I think it is better to install it to `$out/${python.sitePackages}/klayout` with the compile options or move it in the postInstall phase.

Reply:
This is not possible as `klayout` is not a python package, but a GTK application providing a python interface, and it expects its python modules to be at that location.

Reply:
Then use `buildPythonApplication` instead?

Reply:
`kfactory` depends on `klayout`, [registered in `PyPI`](https://pypi.org/project/klayout/).
So `klayout` is a python library here, isn't it?

Reply:
No, it's the full application.

Reply:
https://www.klayout.de/

Reply:
It is common for applications to have python bindings.
For example, kicad is a similar application.

Reply:
You're right. There is a standalone module and it works with `kfactory`. I'll package it and push. Thanks.

Reply:
@natsukium I've implemented the change that you requested, but I see two disadvantages. 
* If you install `kfactory` it's because you want to work wiith `klayout`. By installing the separate python interface you end up with 2 copies of several library files taking unnecessary space.
* In addition, for the above reason, there's no guarantee that the two versions will always stay in sync.

Symlinks are heavily use in nixpkgs. Just take all the `.withPackages` implementations as examples.

Anyway, I made as you wish.